src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -864,6 +864,16 @@ sealed class RsDiagnostic(
         }
     }
 
+    class ImplForNonAdtError(
+        element: PsiElement
+    ) : RsDiagnostic(element) {
+        override fun prepare() = PreparedAnnotation(
+            ERROR,
+            E0118,
+            "Can impl only `struct`s, `enum`s and `union`s"
```

Comment:
The message differs from the compiler one: `error[E0118]: no base type found for inherent implementation`

Reply:
I thought this message might be somewhat simpler to understand than the compiler's one and seeing how other error messages don't follow compiler error messages that closely I thought it'd be okay. I can change it to the compiler's message though no problem.

Reply:
Okaay, sounds reasonable